scripts/nb-tester/qiskit_docs_notebook_tester/__init__.py

```diff
@@ -197,27 +179,39 @@ async def execute_notebook(path: Path, args: argparse.Namespace, config: Config)
     print(f"✅ No problems in {path} (written)")
     return True
 
-async def _execute_notebook(filepath: Path, args: argparse.Namespace) -> nbformat.NotebookNode:
+async def _execute_notebook(filepath: Path, args: argparse.Namespace, patch: bool) -> nbformat.NotebookNode:
     """
-    Use nbconvert to execute notebook.
+    Use nbclient to execute notebook. The steps are:
+    1. Read notebook from file
+    2. Create a new kernel
+    3. (Optional) Run some custom code to set up the kernel
+    4. Execute the notebook inside the kernel
+    5. Clean the notebook and return it
```

Comment:
This is great to list out the steps! Imo no need to duplicate it here and inline, though. It risks going out of sync. The inline comments are better